### DISCUSSION TITLE
Concurrency with automatic retries?

### DISCUSSION THREAD

**Discussion by boryn** [saloonphp/saloon#335] [Q&A]
Hello!

I wonder if there is any way to implement automatic retries for the concurrent requests?

On the one hand it's written in the docs that "_The retry functionality will only work when sending requests synchronously._" 

On the other hand, with asynchronous requests there is `exceptionHandler` for handling the failed requests. So maybe there would be some way to combine both? If it fails, just let it somehow run again?

**Comment by Sammyjo20** [comment]
Hey @boryn! Looks like someone has already got you covered. We're working on adding support for asynchronous requests in #323
